dataset/filter.py

```diff
@@ -0,0 +1,370 @@
+        file2_range = (first[3], last[4])
+
+        old_file_start = min(file1_range[0], old_file_start)
+        old_file_end = max(file1_range[1], old_file_end)
+
+        new_file_start = min(file2_range[0], new_file_start)
+        new_file_end = max(file2_range[1], new_file_end)
+
+    # -2 for compatibility with gh_diff
+    example["old_change_start"] = old_file_start
+    example["old_change_end"] = old_file_end
+    example["old_change_range"] = old_file_end - old_file_start
+
+    example["new_change_start"] = new_file_start
+    example["new_change_end"] = new_file_end
+    example["new_change_range"] = new_file_end - new_file_start
+
+    example["n_inserts"] = n_inserts
+    example["n_deletes"] = n_deletes
+    example["n_changes"] = n_inserts + n_deletes
+
+    return example
+'''
+
+def clean_issues_and_refs(example):
+    """
+    Remove first word if 
+        - [ ] in first word
+        - : in first word
+
+    Remove final word if
+        - [ ] in final word
+        - (# ) in final word
+
+    E.g. 
+    - [benchmark] Fix billing project (#9671) -> Fix billing project
+    - demo/python/cmd.py: Fix struct.unpack format for Python 3 -> Fix struct.unpack format for Python 3
+    """
+    if len(example["subject"]) == 0:
+        return example
+
+    subject = example["subject"].split() + ["", "", ""] # add empty strings to avoid index out of range
+
+    if subject[0].startswith("[") and subject[0].endswith("]"):
+        subject = subject[1:]
+
+    if subject[0].endswith(":"):
+        subject = subject[1:]
+
+    if subject[-1].startswith("[") and subject[-1].endswith("]"):
+        subject = subject[:-1]
+
+    if "#" in subject[-1]:
+        subject = subject[:-1]
+
+    example["subject"] = " ".join(subject).strip()
+
+    return example
+
+# Did not improve results
+# ds = ds.filter(lambda x: x["proba"] >= 0.9, num_proc=30)
+# print("After proba filtering, the dataset size is: {}".format(len(ds)))
+
+ds = ds.filter(lambda x: len(x["old_contents"]) < 100_000, num_proc=NUM_PROC)
+
+print("After content length filtering, the dataset size is: {}".format(len(ds)))
+
+ds = ds.filter(lambda x: len(x["new_contents"]) != 0, num_proc=NUM_PROC)
+
+print("After empty new content filtering, the dataset size is: {}".format(len(ds)))
+
+ds = ds.filter(lambda x: x["old_contents"] != x["new_contents"], num_proc=NUM_PROC)
+
+print("After content equality filtering, the dataset size is: {}".format(len(ds)))
+
+def filter_empty_messages(example):
+    # Require a minimum of 4 chars (even in chinese you can't really write an instruction with <4 chars)
+    # The git commits data should already be filtered for messages with len > 5
+    # Only filter out single alphabetic words (i.e. leave in e.g. Chinese)
+    if len(example["subject"]) < 4 or (len(example["subject"].split()) == 1 and example["subject"].isalpha()):
+        return False
+    return True
+
+ds = ds.filter(filter_empty_messages, num_proc=NUM_PROC)
```

Comment:
Why we pass by the function `filter_empty_messages` two times? I think we can remove this one right?

Reply:
Okay I see.